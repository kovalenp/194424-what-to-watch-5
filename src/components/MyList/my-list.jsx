import React from "react";
import { Link } from "react-router-dom";

import Footer from "../Footer/footer";
import MoviesList from "../MoviesList/movies-list";
import { moviesProps } from "../../validation/propTypes";

const MyList = ({ movies }) => (
  <>
    <div className="user-page">
      <header className="page-header user-page__head">
        <div className="logo">
          <Link to="/" className="logo__link">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <h1 className="page-title user-page__title">My list</h1>

        <div className="user-block">
          <div className="user-block__avatar">
            <img
              src="img/avatar.jpg"
              alt="User avatar"
              width="63"
              height="63"
            />
          </div>
        </div>
      </header>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>
        <MoviesList movies={movies} />
      </section>

      <Footer />
    </div>
  </>
);

MyList.propTypes = moviesProps;

export default MyList;
