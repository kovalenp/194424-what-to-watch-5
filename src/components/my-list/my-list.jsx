import React, {useEffect} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {pullMyFavs} from "../../services/user-service";

import Header from "../header/header";
import Footer from "../footer/footer";
import MoviesList from "../movies-list/movies-list.jsx";
import {movieProps} from "../../validation/propTypes";

const MyList = (props) => {

  useEffect(() => {
    props.getMyFavs();
  }, []);

  return (
    <>
      <div className="user-page">
        <Header pageTitle="My List" />
        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>
          <MoviesList movies={props.myFavorites} />
        </section>

        <Footer />
      </div>
    </>);

};

MyList.propTypes = {
  myFavorites: PropTypes.arrayOf(movieProps),
  getMyFavs: PropTypes.func,
};

const mapStateToProps = (state) => {
  return {
    myFavorites: state.USER.favorites
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    getMyFavs: () => dispatch(pullMyFavs()),
  };
};

export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);
