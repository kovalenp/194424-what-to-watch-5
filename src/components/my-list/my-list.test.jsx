import React from "react";
import renderer from "react-test-renderer";
import {MyList} from "./my-list";
import movie from "../../mocks/movie";

jest.mock(`../header/header`, () => ({
  __esModule: true,
  default: () => () => <p>Header</p>,
}));

jest.mock(`../movies-list/movies-list`, () => ({
  __esModule: true,
  default: () => () => <p>Hey-Im-List</p>,
}));

describe(`MyList meets B21`, () => {

  afterAll(() => {
    jest.clearAllMocks();
  });

  it(`and rendered correctly`, () => {
    const tree = renderer.create((
      <MyList myFavorites={[]} getMyFavs={() => [movie]} />
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
