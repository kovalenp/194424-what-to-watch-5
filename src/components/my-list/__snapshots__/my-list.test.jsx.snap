// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList meets B21 and rendered correctly 1`] = `
<div
  className="user-page"
>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
